test/stability/testMerge/testMerge.go

```diff
@@ -89,151 +123,440 @@ func main() {
-	if err := executeTest(*configFile, *target, testScenario); err != nil {
-		fmt.Printf("fail: %+v\n", err)
+	flow.CheckFatal(initDPDK())
+	if useSimpleGenerate {
+		generatorType = generate
+	} else {
+		generatorType = fastGenerate
 	}
+	flow.CheckFatal(executeTest(*configFile, *target, testScenario, pipelineScenario(pipelineType), generatorType, addSegmAfterMerge))
 }
 
-func executeTest(configFile, target string, testScenario uint) error {
-	if testScenario > 3 || testScenario < 0 {
-		return errors.New("testScenario should be in interval [0, 3]")
-	}
-	// Init NFF-GO system
-	config := flow.Config{
-		DPDKArgs: []string{dpdkLogLevel},
+// getter - is a part of pipeline getting packets by receive or generate or fast generate
+type getScenario uint
+
+const (
+	generate getScenario = iota
+	fastGenerate
+	recv
+)
+
+func setGetter(scenario getScenario, inputSource uint) (finalFlow *flow.Flow, err error) {
+	switch scenario {
+	case recv:
+		finalFlow, err = flow.SetReceiver(uint16(inputSource))
+	case fastGenerate:
+		if inputSource == useGenerator1 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
+	default:
+		if inputSource == useGenerator1 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
 	}
-	if err := flow.SystemInit(&config); err != nil {
-		return err
+	return finalFlow, err
+}
+
+// segment type definition, not all possible values are used to decrease complexity
+type segmentScenario uint
+
+const (
+	handle segmentScenario = iota
+	handleDrop
+	split
+)
+
+// returns nil if no new flow was created and new flow if was
+func setSegment(scenario segmentScenario, f *flow.Flow) (finalFlow *flow.Flow, err error) {
+	err = errors.New("Unrecognized configuration of segment part")
+	switch scenario {
+	case handle:
+		err = flow.SetHandler(f, handlePackets, nil)
+	case handleDrop:
+		err = flow.SetHandlerDrop(f, filterPackets, nil)
+	case split:
+		var outputFlows []*flow.Flow
+		outputFlows, err = flow.SetSplitter(f, splitPackets, uint(2), nil)
+		if err != nil {
+			return nil, err
+		}
+		finalFlow = outputFlows[1]
+		*f = *outputFlows[0]
 	}
-	stabilityCommon.InitCommonState(configFile, target)
+	return finalFlow, err
+}
+
+func splitPackets(currentPacket *packet.Packet, context flow.UserContext) uint {
+	return 0
+}
+
+func handlePackets(pkt *packet.Packet, context flow.UserContext) {
+}
+
+func filterPackets(pkt *packet.Packet, context flow.UserContext) bool {
+	return true
```

Comment:
Also all packets to one part. Remove this. Handle and partition is quite enough.